daemon/containerd/service.go

```diff
@@ -74,6 +78,16 @@ func NewService(config ImageServiceConfig) *ImageService {
 	}
 }
 
+func (i *ImageService) snapshotterService(snapshotter string) snapshots.Snapshotter {
+	s, ok := i.snapshotterServices[snapshotter]
+	if !ok {
+		s = i.client.SnapshotService(snapshotter)
+		i.snapshotterServices[snapshotter] = s
+	}
+
+	return s
```

Comment:
For posterity / future self; I _think_ this is fine to do, but technically this map would be "unbounded", so would allow an "unlimited" number of snapshotters to be referenced here.

I don't think there's any realistic security here (unless someone would want to DOS themselves by having a gazillion snapshotters).